### PR TITLE
Set zero (infinite) options timeout in case of Timeout.InfiniteTimeSpan

### LINKED ISSUE #55984: QUIC: handling of timeouts should be improved
Currently we set timeout in SafeMsQuicConfigurationHandle

https://github.com/dotnet/runtime/blob/633e04481f21a23d567cf49177f76fbef3a34cfe/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs#L118-L127

When timeout _is_ Timeout.InfiniteTimeSpan we should set the value to 0. 
If we skip setting it, default 30s timeout would still apply. 

There is _separate_ timeout setting for Connect. So if somebody wishes to control that either by allowing more time on slow network or vice versa they currently cannot do that via timeouts. (smaller timeout can be enforced via cancellation token as mitigation ) We can either apply current timeout in both cases or split the timeout setting.

### PR DESCRIPTION
**Pull request by Sundow**
Fixes #55984

### CODE REVIEW

**Review by ManickaP — APPROVED** [review]
LGTM, thanks for the contribution!

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs [hunk 1]
```diff
@@ -156,9 +156,13 @@ private static unsafe SafeMsQuicConfigurationHandle Create(QuicOptions options,
                 ulong ms = (ulong)options.IdleTimeout.Ticks / TimeSpan.TicksPerMillisecond;
                 if (ms > (1ul << 62) - 1) throw new Exception("IdleTimeout is too large (max 2^62-1 milliseconds)");
 
-                settings.IsSetFlags |= QuicSettingsIsSetFlags.IdleTimeoutMs;
                 settings.IdleTimeoutMs = (ulong)options.IdleTimeout.TotalMilliseconds;
             }
+            else
+            {
+                settings.IdleTimeoutMs = 0;
```
Comment (stephentoub) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:163]: msquic itself uses 0 to mean infinite?

Comment (Sundow) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:163]: This is what I understand from #55984 explanation.

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:163]: Yep, 0 mean infinite: https://github.com/microsoft/msquic/blob/4b41f2469b68d2fea0682d6f68854399a239b5bf/docs/api/QUIC_SETTINGS.md#members
> How long a connection can go idle before it is gracefully shut down. 0 to disable timeout.

Comment (stephentoub) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:163]: Thanks.  Out of curiosity, how do you say "don't let connections go idle at all"?  I'd have expected that to be what 0 meant.  Is there a separate "disable pooling entirely" setting or something, or is that just not a thing at all?

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:163]: I'm not sure if I understand what you mean by "don't let connections go idle at all". Do you mean something like keep alive?

Comment (ManickaP) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:163]: > EDIT: Ah, this isn't about pooling, it's about how long a connection in use can go without data being transferred?

yep, it's basically auto-close after a time.

**Comment by Sundow** [comment]
3 checks out of 127 failed/cancelled due to a server timeout. Not the first time I see it. Do I need to do anything about it on my side?

**Comment by ManickaP** [comment]
Not really. Your merge triggered another run of CI so I'll wait for that and if there are not failures related to QUIC/HTTP3, I'll merge it.

**Pull request merged**